anthos-onprem-terraform/avmw_user_cluster_control_plane_v2/variables.tf

```diff
@@ -0,0 +1,161 @@
+  description = "The name of the user cluster to be created"
+  type        = string
+  default     = "vmware-metallb-user-cluster"
+}
+
+variable "admin_cluster_name" {
+  description = <<EOT
+    The name of the admin cluster that manages the user cluster. The admin cluster
+    name is the last segment of the fully-specified cluster name that uniquely
+    identifies the cluster in Google Cloud:
+    projects/FLEET_HOST_PROJECT_ID/locations/global/memberships/ADMIN_CLUSTER_NAME
+  EOT
+  type        = string
+}
+
+variable "on_prem_version" {
+  description = <<EOT
+     The Anthos clusters on the VMware version for your user cluster.
+     Defaults to the admin cluster version.
+  EOT
+  type        = string
+}
+
+variable "control_plane_vip" {
+  description = <<EOT
+    The virtual IP address (VIP) that you have chosen to configure on the load
+    balancer for the Kubernetes API server of the user cluster.
+  EOT
+  type        = string
+  validation {
+    condition     = can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", var.control_plane_vip))
+    error_message = "Invalid IP address for Control plane VIP."
+  }
+}
+
+variable "ingress_vip" {
+  description = <<EOT
+    The IP address that you have chosen to configure on the load balancer for
+    the ingress proxy.
+  EOT
+  type        = string
+  validation {
+    condition     = can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", var.ingress_vip))
+    error_message = "Invalid IP address for Ingress VIP."
+  }
+}
+
+variable "control_plane_node_cpus" {
+  description = <<EOT
+    The number of CPUs for each admin cluster node that serve as control planes
+    for this VMware user cluster.
+  EOT
+  type        = number
+  default     = 4
+}
+
+variable "control_plane_node_memory" {
+  description = <<EOT
+    The megabytes of memory for each admin cluster node that serves as a
+    control plane for this VMware user cluster.
+  EOT
+  type        = number
+  default     = 8192
+}
+
+variable "control_plane_node_replicas" {
+  description = <<EOT
+    The number of control plane nodes for this VMware user cluster.
+  EOT
+  type        = number
+  default     = 3
+}
+
+variable "control_plane_hostname" {
+  description = <<EOT
+    The hostname that you have chosen to configure on control plane.
+  EOT
+  type        = string
+}
+
+variable "control_plane_ip" {
+  description = <<EOT
+    The IP address that you have chosen to configure on control plane.
+  EOT
+  type        = string
+  validation {
+    condition     = can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", var.ingress_vip))
```

Comment:
nit: consider adding to the description the constraint on the IP that the regex validates.